### PR TITLE
Ahs 2021 2022

### PR DESCRIPTION
**Pull request by b47huang**
Change Log
[AHS UWPath.xlsx](https://github.com/UW-Path/DataParsing/files/7109015/AHS.UWPath.xlsx)

Keep additions:
Fixed issues with line using "One of" or "Two of" keywords.

### CODE REVIEW

File: ProgramParsing/AHS/MajorParser2021_2022.py [hunk 1]
```diff
@@ -38,10 +38,16 @@ def _get_relatedMajor(self, program):
...
         # Get courses from line
         line = line.replace("*", "").replace("/", " or ")
+
         coursesA = set(re.findall(r"[A-Z]{2,10}(?:\s[1-9][0-9][0-9][A-Z]?)?(?:\sor\s[A-Z]{2,10}(?:\s[1-9][0-9][0-9][A-Z]?)?){0,10}", line))
         coursesB = set(re.findall(r"(?:[A-Z]{2,10}\s)?[1-9][0-9][0-9][A-Z]?(?:\sor\s[A-Z]{2,10}(?:\s[1-9][0-9][0-9][A-Z]?)?){0,10}", line))
         courses = coursesA.union(coursesB)
         courses = list(courses)
+
+        if ("One" in line and "from the following list" in line) or \
+                "One of" in line or "Two of" in line:
```
Comment (haowei99) [inline ProgramParsing/AHS/MajorParser2021_2022.py]: use regex here: 

if ("One" in line and "from the following list" in line) or re.findall(r"(One|Two|Three|Four|Five|Six|Seven|Eight|Nine|Ten|All) of", line):

Comment (haowei99) [reply ProgramParsing/AHS/MajorParser2021_2022.py]: also, if this is two+ of , are you sure it is [" or ".join(courses)] not [", ".join(courses)]. Do we even have or courses in the DB

File: ProgramParsing/AHS/MajorParser2021_2022.py [hunk 2]
```diff
@@ -129,6 +134,9 @@ def load_file(self, file, year):
                         courses = self.__get_courses(information[i])
                         all_courses = all_courses.union(courses)
 
+                        if "Two of" in information[i]:
```
Comment (haowei99) [inline ProgramParsing/AHS/MajorParser2021_2022.py]: This is too hardcoded - If it is one of two of three of - use StringToNumber.py module to convert such that it is `StringToNumber[{use the found regex number}].value`. You can search this up to see how it is used on the other code base

File: ProgramParsing/AHS/ParseProgram.py [hunk 3]
```diff
@@ -11,9 +11,12 @@
...
+    # for testing purposes
+    # files = set(["/Specs/2021-2022-HEA-Health-Informatics-Option.html"])
+
     parsers = {
         'MajorParser': AHSMajorParser,
         'MajorParser2021_2022': AHSMajorParser2021_2022
     }
 
-    main(parsers, files, "Applied Health Science")
+    main(parsers, files, "Applied Health Science", True)
```
Comment (haowei99) [inline ProgramParsing/AHS/ParseProgram.py]: I don't think you want to drop here right :p

**Review by haowei99 — CHANGES_REQUESTED** [review]
left some comments

File: ProgramParsing/AHS/MajorParser2021_2022.py (hunk 1 above)
Comment (b47huang) [reply ProgramParsing/AHS/MajorParser2021_2022.py]: for the second part with the "or" stuff, theres some weird stuff going on in that Parser, so when we use or, it gets converted to 1 entry into the database, whereas if we use a comma, it enters them as separate entries

Comment (haowei99) [reply ProgramParsing/AHS/MajorParser2021_2022.py]: I think I know what you mean now

**Review by haowei99 — APPROVED** [review]

**Pull request merged**